### DISCUSSION TITLE
Resteasy and Jersey conflict

### DISCUSSION THREAD

**Discussion by ajit2548** [resteasy/resteasy#3851] [Q&A]
while upgrading resteasy-core from v4.7.7 to v6.2.4-Final, i get following error:

java.lang.RuntimeException: java.lang.ClassCastException: class org.glassfish.jersey.uri.internal.JerseyUriBuilder cannot be cast to class org.jboss.resteasy.spi.ResteasyUriBuilder (org.glassfish.jersey.uri.internal.JerseyUriBuilder and org.jboss.resteasy.spi.ResteasyUriBuilder are in unnamed module of loader 'app')

while investigating, i found following code of mine which fails. it fails in 'setUri' method.

ResteasyUriInfo resteasyUriInfo = (ResteasyUriInfo) session.getContext().getHttpRequest().getUri();
        resteasyUriInfo.setUri(originalUriInfo.getBaseUri(), relativePath);
Please guide here.

@jamezp this exception is part of a junit jupiter test(v5.9.3) running in local intellij IDEA. I hope to answer you its a test container.

**Comment by jamezp** [comment]
@ajit2548 It's likely not wise to attempt to use two different Jakarta REST implementations. Can you remove Jersey from your class path?
